### ISSUE TITLE
julia_13 doesn't build

### ISSUE THREAD

**Issue by olynch** [NixOS/nixpkgs#82008]
**Describe the bug**
I haven't tried to build it on my local machine (because it would take forever), but it doesn't build on hydra: https://hydra.nixos.org/build/113989863/nixlog/1/tail. I don't know if people are working on this; is there current information that I could use to debug this?

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- julia_13
# a list of nixos modules affected by the problem
module:
```

**Comment by ivan-timokhin** [comment]
I have tried reproducing it locally, and, curiously, while I can reproduce it via `nix-build`, I *can't* reproduce it via `nix-shell --pure --command genericBuild`, which I tried per the [advice from the wiki](https://nixos.wiki/wiki/Nixpkgs/Create_and_debug_packages) (I get additional libgit2-related errors, but those are a known upstream issue with [patch available](https://github.com/JuliaLang/julia/pull/35232)). So whatever is causing these issues appears to be sensitive to where (and by whom) exactly the package is being built.

**Comment by ivan-timokhin** [comment]
Pinging maintainers: @7c6f434c @rbvermaa @garrison 

In the meanwhile, Julia 1.4 is out, but my attempt to update the expression led to the same build errors, so I would suspect the same cause.

**Comment by 7c6f434c** [comment]
Does it also kind-of-work if you disable the tests, like 1.3?

**Comment by ivan-timokhin** [comment]
No; it complains about missing `libLLVM-8jl.so` which, apparently, doesn't get installed with the rest of the libraries for some reason (the install directory does contain `libLLVM.so` and `libLLVM-8.0.1jl.so`, both hanging symlinks to the same missing library). I'll see if I can just copy it over manually in install phase.

**Comment by olynch** [comment]
For those who are frustrated by this, dropping in
      (stdenv.mkDerivation {
            name = "julia";
            src = pkgs.fetchurl {
                  url = "https://julialang-s3.julialang.org/bin/linux/x64/1.4/julia-1.4.1-linux-x86_64.tar.gz";
                  sha256 = "02k1a17lw7jdzc1ngbr6z6lqs3ivg8h95yxf7i61fy6nmsnqqvgx";
            };
            installPhase = ''
              mkdir $out
              cp -R * $out/
            '';
            dontStrip = true;
      })
into the packages list of a fully-loaded FHS works. I run julia in a FHS anyways, because libraries need it, so it's not a big drawback.

**Comment by rikhuijzer** [comment]
@olynch That sounds very interesting. I was not able to run Julia 1.4 based on your instructions. Do you happen to have a link to a good reference, or even your .nix files?

**Comment by samuela** [comment]
I just came across this as well. Here's my complete log: https://gist.github.com/samuela/a04e336f8fb89926517acb5888c2c6d4.

Are there any updates on getting julia to build?

**Comment by 7c6f434c** [comment]
>Are there any updates on getting julia to build?

If you are willing no to have the tests run automatically during the build, you can disable the check phase and the build itself works (apparently) fine. 

Many of the failing tests are known upstream to be kind of suboptimally done but happen to run on the upstream testing infrastructure fine. I think we never got completely sure if it is all of the failing tests, though.

**Comment by samuela** [comment]
> Many of the failing tests are known upstream to be kind of suboptimally done but happen to run on the upstream testing infrastructure fine. I think we never got completely sure if it is all of the failing tests, though.

Oh interesting. Have we filed issues on the julia project for these failures? I'd be interested in working towards getting a working julia 1.5 derivation up.

**Comment by 7c6f434c** [comment]
>Oh interesting. Have we filed issues on the julia project for these failures? I'd be interested in working towards getting a working julia 1.5 derivation up.

Nope, there were already some…

Probably some of the issues were fixed after 1.5
